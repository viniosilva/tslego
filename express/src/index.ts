import { apiConfig } from './common/config';
import { logger } from './common/logger';
import { Server } from './server';
import { userController } from './user';

const routes = [userController.router];
const server = new Server(
  apiConfig.host,
  apiConfig.port,
  logger,
  apiConfig.swaggerConfig,
  routes,
);

server.start();
