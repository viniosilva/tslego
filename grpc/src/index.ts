import { apiConfig } from './common/config';
import { logger } from './common/logger';
import { Server } from './Server';
import { userHandler } from './user';

const handlers = [userHandler];

const server = new Server(apiConfig.host, apiConfig.port, handlers, logger);
server.start();
