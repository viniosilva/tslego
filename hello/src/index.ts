class Hello {
  getMessage(): string {
    return 'Hello world! (:';
  }
}

export const hello = new Hello();

console.log(hello.getMessage());
