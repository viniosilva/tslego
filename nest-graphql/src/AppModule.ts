import { Module } from '@nestjs/common';
import { GraphQLModule } from '@nestjs/graphql';
import { LoggerModule } from 'nestjs-pino';
import { httpLogger } from './common/httpLogger';
import { UserModule } from './user/UserModule';

@Module({
  imports: [
    LoggerModule.forRoot({ pinoHttp: httpLogger }),
    UserModule,
    GraphQLModule.forRoot({
      autoSchemaFile: 'src/schema.gql',
      sortSchema: true,
    }),
  ],
})
export class AppModule {}
