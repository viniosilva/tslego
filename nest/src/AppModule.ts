import { Module } from '@nestjs/common';
import { LoggerModule } from 'nestjs-pino';
import { httpLogger } from './common/httpLogger';
import { UserModule } from './user/UserModule';

@Module({
  imports: [LoggerModule.forRoot({ pinoHttp: httpLogger }), UserModule],
})
export class AppModule {}
