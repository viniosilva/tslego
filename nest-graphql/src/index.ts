import { ValidationPipe } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';
import helmet from 'helmet';
import { AppModule } from './AppModule';
import { apiConfig } from './common/config';
import { logger } from './common/logger';

async function bootstrap() {
  const app = await NestFactory.create(AppModule, { logger: false });
  try {
    app.useGlobalPipes(new ValidationPipe({ transform: true }));
  } catch (err) {
    logger.error(err);
  }
  // app.use(helmet());

  await app.listen(apiConfig.port);

  logger.info(`App listening at [${apiConfig.host}:${apiConfig.port}]`);
}
bootstrap();
