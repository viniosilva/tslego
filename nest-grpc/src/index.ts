import { NestFactory } from '@nestjs/core';
import { MicroserviceOptions, Transport } from '@nestjs/microservices';
import { AppModule } from './AppModule';
import { apiConfig } from './common/config';
import { logger } from './common/logger';

async function bootstrap() {
  const url = `${apiConfig.host}:${apiConfig.port}`;
  const app = await NestFactory.createMicroservice<MicroserviceOptions>(
    AppModule,
    {
      logger: false,
      transport: Transport.GRPC,
      options: {
        package: 'user',
        protoPath: 'protos/user.proto',
        url,
      },
    },
  );

  app.listen(() => logger.info(`App listening at [${url}]`));
}

bootstrap();
