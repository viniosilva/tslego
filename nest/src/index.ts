import { description, version } from '../package.json';
import { NestFactory } from '@nestjs/core';
import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
import { Logger } from 'nestjs-pino';
import helmet from 'helmet';
import { AppModule } from './AppModule';
import { apiConfig } from './common/config';
import { INestApplication, ValidationPipe } from '@nestjs/common';

async function bootstrap() {
  const app = await NestFactory.create(AppModule, { logger: false });
  app.useGlobalPipes(new ValidationPipe({ transform: true }));
  app.use(helmet());

  configureSwagger(app);

  await app.listen(apiConfig.port);

  const logger = app.get(Logger);
  logger.log(`App listening at [${apiConfig.host}:${apiConfig.port}]`);
}

function configureSwagger(app: INestApplication) {
  const options = new DocumentBuilder()
    .setTitle('TSLego Nest Hello')
    .setDescription(description)
    .setVersion(version)
    .addTag('User', 'Operations about users')
    .build();
  const document = SwaggerModule.createDocument(app, options);
  SwaggerModule.setup('/api/docs', app, document);
}
bootstrap();
