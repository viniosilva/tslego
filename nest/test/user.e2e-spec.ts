import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication, ValidationPipe } from '@nestjs/common';
import request, { Response } from 'supertest';
import { AppModule } from '../src/AppModule';

describe('AppController (e2e)', () => {
  const path = '/api/users';
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    app.useGlobalPipes(new ValidationPipe({ transform: true }));
    await app.init();
  });

  describe('GET /api/users', () => {
    it('should return empty user list', () => {
      return request(app.getHttpServer())
        .get(path)
        .expect(200)
        .expect((res: Response) => {
          expect(res.text).toMatchSnapshot();
        });
    });

    it('should return user list', async () => {
      await request(app.getHttpServer()).post(path).send({ name: 'Test name' });

      return request(app.getHttpServer())
        .get(path)
        .expect(200)
        .expect((res: Response) => {
          expect(res.text).toMatchSnapshot();
        });
    });
  });

  describe('GET /api/users/:userId', () => {
    it('should return user', async () => {
      await request(app.getHttpServer()).post(path).send({ name: 'Test name' });

      return request(app.getHttpServer())
        .get(`${path}/1`)
        .expect(200)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw not found exception', () => {
      return request(app.getHttpServer())
        .get(`${path}/0`)
        .expect(404)
        .expect((res: Response) => {
          expect(res.text).toMatchSnapshot();
        });
    });
  });

  describe('POST /api/users', () => {
    it('should return new user', () => {
      const userInput = { name: 'Test name' };
      return request(app.getHttpServer())
        .post(path)
        .send(userInput)
        .expect(201)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw bad request exception', () => {
      const userInput = { name: 123 };
      return request(app.getHttpServer())
        .post(path)
        .send(userInput)
        .expect(400)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });

  describe('PUT /api/users/:userId', () => {
    it('should return updated user', async () => {
      await request(app.getHttpServer()).post(path).send({ name: 'Test name' });

      const userInput = { name: 'Test update name' };
      return request(app.getHttpServer())
        .put(`${path}/1`)
        .send(userInput)
        .expect(200)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw bad request exception', () => {
      const userInput = { name: 123 };
      return request(app.getHttpServer())
        .put(`${path}/0`)
        .send(userInput)
        .expect(400)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw not found exception', () => {
      const userInput = { name: 'Test update name' };
      return request(app.getHttpServer())
        .put(`${path}/0`)
        .send(userInput)
        .expect(404)
        .expect((res: Response) => {
          expect(res.text).toMatchSnapshot();
        });
    });
  });

  describe('DELETE /api/users/:userId', () => {
    it('should remove user', async () => {
      await request(app.getHttpServer()).post(path).send({ name: 'Test name' });
      return request(app.getHttpServer()).delete(`${path}/1`).expect(204);
    });
  });
});
