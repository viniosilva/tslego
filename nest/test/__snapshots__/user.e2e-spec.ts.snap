// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) GET /api/users should return empty user list 1`] = `"[]"`;

exports[`AppController (e2e) GET /api/users should return user list 1`] = `"[{\\"id\\":1,\\"name\\":\\"Test name\\"}]"`;

exports[`AppController (e2e) GET /api/users/:userId should return user 1`] = `
Object {
  "id": 1,
  "name": "Test name",
}
`;

exports[`AppController (e2e) GET /api/users/:userId should throw not found exception 1`] = `"{\\"statusCode\\":404,\\"message\\":\\"User not found\\",\\"error\\":\\"Not Found\\"}"`;

exports[`AppController (e2e) POST /api/users should return new user 1`] = `
Object {
  "id": 1,
  "name": "Test name",
}
`;

exports[`AppController (e2e) POST /api/users should throw bad request exception 1`] = `
Object {
  "error": "Bad Request",
  "message": Array [
    "name must be longer than or equal to 3 and shorter than or equal to 100 characters",
    "name must be a string",
  ],
  "statusCode": 400,
}
`;

exports[`AppController (e2e) PUT /api/users/:userId should return updated user 1`] = `
Object {
  "id": 1,
  "name": "Test update name",
}
`;

exports[`AppController (e2e) PUT /api/users/:userId should throw not found exception 1`] = `"{\\"statusCode\\":404,\\"message\\":\\"User not found\\",\\"error\\":\\"Not Found\\"}"`;
